core/frontend/src/tile/ImdlReader.ts

```diff
@@ -142,12 +142,23 @@ interface ImdlTextureMapping {
     mode?: TextureMapping.Mode;
     /** @see [TextureMapping.Params.worldMapping]($common). Default: false. */
     worldMapping?: boolean;
+    /** @see [TextureMapping.Params.useConstantLod]($common). Default: false. */
+    useConstantLod?: boolean;
+    /** Describes the [TextureMapping.ConstantLodParamProps]($common). */
+    constantLodParams?: {
```

Comment:
Why isn't this just `constantLodParams?: ConstantLodParamProps;`?

Reply:
You mean why do we need the boolean?  Because if the normal map wants to use constant lod mapping but the texture doesn't you need a flag somewhere.  You can't just go by the existance of the constantLodParams.

Reply:
No, you defined an interface type inline for `constantLodParams` that looks (to me) identical to the definition of `ConstantLodParamProps`.

Reply:
Ahh, they're a little bit different in the offset.  In this case offset is an array of 2 numbers while it is an XAndY in ConstantLodParamProps.  I didn't see anything like that here (all other interfaces defined at the top of ImdlReader.ts for this purpose use just arrays of numbers.  The transform for instance is also just defined as number[][]).